controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1204,6 +1204,20 @@ func (r *BareMetalHostReconciler) getBMCSecretAndSetOwner(request ctrl.Request,
 	return bmcCredsSecret, nil
 }
 
+func credentialsFromSecret(bmcCredsSecret *corev1.Secret) *bmc.Credentials {
+	// We trim surrounding whitespace because those characters are
+	// unlikely to be part of the username or password and it is
+	// common for users to encode the values with a command like
+	//
+	//     echo "my-password" | base64
+	//
+	// which introduces a trailing newline.
+	return &bmc.Credentials{
+		Username: strings.TrimSpace(string(bmcCredsSecret.Data["username"])),
```

Comment:
Note: it looks like that `TrimSpace` consider the Unicode set of characters as a "space": https://golang.org/pkg/unicode/#IsSpace. For the sake of clarity I'd suggest to document it.

Reply:
That set looks the same as "whitespace" to me. Is it different?

Reply:
No it's the same, so probably not required